packages/common/lib/common/tools.js

```diff
@@ -3222,6 +3222,125 @@ function showDeprecatedMessage(func, logger) {
     logger(`Function "${func}" is deprecated. Please use async version of it: "${func}Async"`);
 }
 
+/**
+ * Requests the licenses from ioBroker.net
+ * @param {string} login Login for ioBroker.net
+ * @param {string} password Decoded password for ioBroker.net
+ * @returns {object[]} array of all licenses stored on iobroker.net
+ */
+function _readLicenses(login, password) {
+    axios = axios || require('axios');
+    const config = {
+        headers: { Authorization: `Basic ${Buffer.from(login + ':' + password).toString('base64')}` },
+        timeout: 4000
+    };
+
+    return axios.get(`https://iobroker.net:3001/api/v1/licenses`, config)
+        .then(response => {
+            if (response.data && response.data.length) {
+                const now = Date.now();
+                response.data = response.data.filter(license => !license.validTill || license.validTill === '0000-00-00 00:00:00' || new Date(license.validTill).getTime() > now);
+            }
+            return response.data;
+        })
+        .catch(err => {
+            if (err.response) {
+                throw new Error((err.response.data && err.response.data.error) || err.response.data || err.response.status);
+            } else if (err.request) {
+                throw new Error('no response');
+            } else {
+                throw err;
+            }
+        });
+}
+
+/**
+ * Reads the licenses from iobroker.net
+ * Reads the licenses from iobroker.net and if no login/password provided stores it in system.licenses
+ * @param {object} objects Object store instance
+ * @param {string} login Login for ioBroker.net
+ * @param {string} password Decoded password for ioBroker.net
+ * @returns {object[]} array of all licenses stored on iobroker.net
+ */
+function updateLicenses(objects, login, password) {
+    // if login and password provided in the message, just try to read without saving it in system.licenses
+    if (login && password) {
```

Comment:
Does that mean that the password comes unencryted here? Should we do that when thinking about Admin (pot. via cloud?)

Reply:
Yes it is unencrypted, but it will be stored encrypted.

Reply:
I do not understand about "Admin (pot. via cloud?)"